diff --git a/src/libnet_pblock.c b/src/libnet_pblock.c
index 9fd015f..fa46d35 100644
--- a/src/libnet_pblock.c
+++ b/src/libnet_pblock.c
@@ -500,10 +500,9 @@ libnet_pblock_coalesce(libnet_t *l, uint8_t **packet, uint32_t *size)
 				q->ptag, libnet_diag_dump_pblock_type(q->type),
 				ip_offset);
 #endif
-                        c = libnet_inet_checksum(l, iph,
-                                libnet_pblock_p2p(q->type), q->h_len,
-                                beg, end);
-                        if (c == -1)
+                        if (libnet_inet_checksum(l, iph,
+                                                 libnet_pblock_p2p(q->type), q->h_len,
+                                                 beg, end) == -1)
                         {
                             /* err msg set in libnet_do_checksum() */
                             goto err;
